### ISSUE TITLE
Brunch

### ISSUE THREAD

**Issue by MajorBreakfast** [stefanpenner/ember-app-kit#296]
TL;DR We should take a look at brunch

First of, here is what Ember App Kit is all about:
- Pipeline for building Ember Apps
  - Ships with the essential tools
  - Other tools like SCSS and CoffeeScript are preconfigured and can be installed in mere seconds
- Boilerplate that encourages a clean app structure and testing
- Intelligent resolver that imports needed modules automatically. We even use the upcoming ES6 module syntax

And here is what we want in addition to that:
- Superfast rebuilding even for large apps
- Some kind of "Ember modules": Written in ES6 style and imported automatically

I think grunt served us very well. Hell, I've spent the past few weeks tweaking our pipeline. Now we've come to the point that we need per file rebuilding. I'm really not sure if we can build such a thing with grunt. It all comes down to the fact that grunt thinks in terms of tasks. What we really need is a tool that looks at each file and reacts in a smart way.

Grunt is like:
"Oh the window is dirty, let's tear down the house and build a new one" - me :)

In the past day I've taken a close look at brunch. The first thing that surprised me was that it's insanely fast at rebulding. And it seems that the brunch people take source maps seriously. That means you can use the chrome dev tool's editing capabilities.

There are brunch based ember project templates out there but none is as ambitious as app kit is. I've already talked to @stefanpenner about this and he isn't against switching. The thing is: This is no incrematal change, it's rather big.

It means:
- Configuring brunch (Shoudn't be too hard)
- Create an ES6 module transpiler brunch plugin. Preferably without breaking sourcemaps
- Getting our `/tests` page working

Why do we have to switch to something else, can't we just improve the setup of grunt's watch task?
The watch task offers a really crappy api for reacting to file changes. It's called the `watch` event and it gets emitted once for each changed file. It is possible to reconfigure the clean task and the respective build task in there. I've tried it and it works kind of. However, I didn't get it working so that it also recognizes the updated compiled file. (compile/copy -> ~~transpile -> concat~~) So that's one problem. AND: It's all very verbose. The watch task is the wrong tool. It's great if all you want is a rough reaction. That's what we have now and it definitly works but it feels sluggish if you do livereloading. We could build a tool that uses `gaze` or `chokidar` for watching the files and reacts in a smarter way when it detects file changes. The thing is that that's more or less exactly what brunch is.

Let's get the discussion started:
Who wants to help? Who thinks that this is stupid? Has anyone a better idea?
